core/src/main/com/github/mmorihiro/core/MakeNewGeneration.kt

```diff
@@ -3,6 +3,48 @@ package com.github.mmorihiro.core
 
 internal class MakeNewGeneration {
     fun get(array: Array<BooleanArray>): Array<BooleanArray> {
-        return array
+        val copy = copy(array)
+        for (i in array.indices) {
+            for (j in array[0].indices) {
+                val pair = Pair(array[i][j], countSurroundings(array, i, j))
+
+                when (pair) {
+                    Pair(false, 3) -> copy[i][j] = true
+                    Pair(true, 2), Pair(true, 3) -> copy[i][j] = true
+                    Pair(true, 0), Pair(true, 1) -> copy[i][j] = false
+                    else ->
+                        if (pair.first == true && pair.second >= 4)
+                            copy[i][j] = false
+                }
+            }
+        }
+
+        return copy
+    }
+
+    private fun copy(array: Array<BooleanArray>): Array<BooleanArray> {
+        val copy = Array(array.size) { BooleanArray(array[0].size) { false } }
+        for (i in array.indices) {
+            for (j in array[0].indices) {
+                copy[i][j] = array[i][j]
+            }
+        }
+        return copy
+    }
```

Comment:
Copying array is very important. Actually, not knowing `Array.copyOf` uses shallow copy, I spent long time.
We have to use deep copy and write the code ourselves.